h/api/nipsa/logic.py

```diff
@@ -0,0 +1,51 @@
+import json
+
+from h.api.nipsa import models
+
+
+def _publish(request, data):
+    request.get_queue_writer().publish("nipsa_user_requests", data)
+
+
+def index():
+    """Return a list of all the NIPSA'd user IDs.
+
+    :rtype: list of unicode strings
+
+    """
+    return [nipsa_user.userid for nipsa_user in models.NipsaUser.all()]
+
+
+def nipsa(request, userid):
+    """NIPSA a user.
+
+    Add the given user's ID to the list of NIPSA'd user IDs.
+    If the user is already NIPSA'd then nothing will happen (but a "nipsa"
+    message for the user will still be published to the queue).
+
+    """
+    nipsa_user = models.NipsaUser.get_by_id(userid)
+    if not nipsa_user:
+        nipsa_user = models.NipsaUser(userid)
+        request.db.add(nipsa_user)
+
+    _publish(request, json.dumps({"action": "nipsa", "userid": userid}))
+
+
+def unnipsa(request, userid):
+    """Un-NIPSA a user.
+
+    If the user isn't NIPSA'd then nothing will happen (but an "unnipsa"
```

Comment:
Well, I guess the point is that a) re-nipsa'ing people doesn't do any harm and b) maybe one day there'll be a reason why we want to re-nipsa someone so why write unnecessary logic that prevents re-nipsa'ing?

This is the same reason why I didn't add logic to prevent you from nipsaing a user id that doesn't exist in the users table - because I can imagine (possibly unlikely) future scenarios where we may _want_ to nipsa users who aren't in the users table, and adding logic to prevent it is just unnecessary